tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/label/LabelRepositoryContract.scala

```diff
@@ -0,0 +1,236 @@
+    val aliceLabel = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED)))).block()
+    val bobLabel = SMono.fromPublisher(testee.addLabel(BOB, LabelCreationRequest(DisplayName("Important"), Some(RED)))).block()
+
+    val labels = SFlux.fromPublisher(testee.getLabels(ALICE, java.util.List.of(aliceLabel.id)))
+      .collectSeq().block()
+
+    assertThat(labels.asJava).containsOnly(aliceLabel)
+  }
+
+  @Test
+  def listShouldReturnEmptyByDefault(): Unit = {
+    val labels = SFlux.fromPublisher(testee.listLabels(ALICE)).collectSeq().block()
+
+    assertThat(labels.asJava).isEmpty()
+  }
+
+  @Test
+  def listShouldReturnAllLabelsOfTheUser(): Unit = {
+    val label1 = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED)))).block()
+    val label2 = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED)))).block()
+
+    val labels = SFlux.fromPublisher(testee.listLabels(ALICE)).collectSeq().block()
+
+    assertThat(labels.asJava).containsExactlyInAnyOrder(label1, label2)
+  }
+
+  @Test
+  def listShouldNotReturnOtherUsersLabels(): Unit = {
+    val aliceLabel = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED)))).block()
+    val bobLabel = SMono.fromPublisher(testee.addLabel(BOB, LabelCreationRequest(DisplayName("Important"), Some(RED)))).block()
+
+    val labels = SFlux.fromPublisher(testee.listLabels(ALICE)).collectSeq().block()
+
+    assertThat(labels.asJava).containsOnly(aliceLabel)
+  }
+
+  @Test
+  def deleteShouldSucceed(): Unit = {
+    val label1 = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED))))
+      .block()
+    val label2 = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED))))
+      .block()
+
+    SMono.fromPublisher(testee.deleteLabel(ALICE, label1.id)).block()
+
+    val labels = SFlux.fromPublisher(testee.listLabels(ALICE)).collectSeq().block()
+    assertThat(labels.asJava).containsOnly(label2)
+  }
+
+  @Test
+  def deleteShouldNotRemoveOtherUsersLabels(): Unit = {
+    val aliceLabel = SMono.fromPublisher(testee.addLabel(ALICE, LabelCreationRequest(DisplayName("Important"), Some(RED))))
+      .block()
+    val bobLabel = SMono.fromPublisher(testee.addLabel(BOB, LabelCreationRequest(DisplayName("Important"), Some(RED))))
+      .block()
+
+    SMono.fromPublisher(testee.deleteLabel(ALICE, aliceLabel.id)).block()
+
+    val labels = SFlux.fromPublisher(testee.listLabels(BOB)).collectSeq().block()
+    assertThat(labels.asJava).containsOnly(bobLabel)
+  }
+
+  @Test
+  def deleteShouldBeIdempotent(): Unit = {
+    val randomLabelId = LabelId.generate()
+
+    assertThatCode(() => SMono.fromPublisher(testee.deleteLabel(ALICE, randomLabelId)).block())
+      .doesNotThrowAnyException()
+  }
+}
```

Comment:
Should we?
- updateShouldNotThrowWhenIndempotent
- updateShouldThrowWhenBotColorAndDisplayNameIsEmpty

Reply:
> updateShouldThrowWhenBothColorAndDisplayNameIsEmpty

Why does it should throw? Maybe just not overriding anything is good enough? e.g: https://github.com/linagora/tmail-backend/pull/724/commits/856c110ab3e0fe8de33106fa8a7d848a9403b0e9

Reply:
that's ok